meson.build

```diff
@@ -0,0 +1,1629 @@
+project('mpv',
+        'c',
+        license: ['GPL2+', 'LGPL2.1+'],
+        version: '0.33',
+        default_options: [
+            'buildtype=debugoptimized',
+            'c_std=c11',
+            'warning_level=1',
+        ]
+)
+
+cc = meson.get_compiler('c')
+
+
+# essential dependencies
+ffmpeg = {
+    'name': 'ffmpeg',
+    'deps': [dependency('libavutil', version: '>= 56.12.100'),
+             dependency('libavcodec', version: '>= 58.12.100'),
+             dependency('libavformat', version: '>= 58.9.100'),
+             dependency('libswscale', version: '>= 5.0.101'),
+             dependency('libavfilter', version: '>= 7.14.100'),
+             dependency('libswresample', version: '>= 3.0.100')],
+}
+
+libass = dependency('libass', version: '>= 0.12.2')
+
+pthreads = dependency('threads')
+
+zlib = cc.find_library('z')
+
+dependencies = [ffmpeg['deps'],
+                libass,
+                pthreads,
+                zlib]
+
+iconv = find_program('iconv', required: get_option('iconv'))
+if not iconv.found() and not get_option('iconv').disabled()
+    error('''Unable to find iconv which should be part of a standard
+          compilation environment. Aborting. If you really mean to compile
+          without iconv support use -Diconv=disabled.''')
+endif
+
+
+# compiler/linker flags
+flags = ['-D_ISOC99_SOURCE', '-D_GNU_SOURCE',
+         '-D_FILE_OFFSET_BITS=64']
+link_flags = []
+
+test_flags = ['-Werror=implicit-function-declaration',
+              '-Wno-error=deprecated-declarations',
+              '-Wno-error=unused-function',
+              '-Wempty-body',
+              '-Wdisabled-optimization',
+              '-Wstrict-prototypes',
+              '-Wno-format-zero-length',
+              '-Werror=format-security',
+              '-Wno-redundant-decls',
+              '-Wvla',
+              '-Wno-format-truncation',
+              '-Wimplicit-fallthrough',
+              '-fno-math-errno']
+
+flags += cc.get_supported_arguments(test_flags)
+
+if cc.get_id() == 'gcc'
+    gcc_flags = ['-Wundef', '-Wmissing-prototypes', '-Wshadow',
+                 '-Wno-switch', '-Wparentheses', '-Wpointer-arith',
+                 '-Wno-pointer-sign',
+                 # GCC bug 66425
+                 '-Wno-unused-result']
+    flags += gcc_flags
+endif
+
+if cc.get_id() == 'clang'
+    clang_flags = ['-Wno-logical-op-parentheses', '-fcolor-diagnostics',
+                   '-Wno-tautological-compare',
+                   '-Wno-tautological-constant-out-of-range-compare']
+    flags += clang_flags
+endif
+
+vector = {
+    'name': 'vector',
+    'statement': run_command('cat', 'waftools/fragments/vector.c').stdout(),
```

Comment:
Use of run_command with cat, is not portable to Windows, which may be annoying when getting the Windows build running.

Worse: it's redundant and pointless. cc.compiles (3 lines down) can accept a `files('waftools/fragments/vector.c')` object and try to compile the file rather than creating a temporary file from a meson string. :)